src/Symfony/Component/Serializer/NameConverter/CamelCaseToSnakeCaseNameConverter.php

```diff
@@ -48,7 +48,7 @@ public function normalize($propertyName)
 
             $len = strlen($propertyName);
             for ($i = 0; $i < $len; ++$i) {
-                if (ctype_upper($propertyName[$i])) {
+                if (ctype_upper($propertyName[$i]) && $i !== 0) {
```

Comment:
Ok, got it. Your improvement looks good!

The code can be still more readable:
```php
$lcPropertyName = strtolower($propertyName[$i]);
if (0 !== $i && $lcPropertyName !== $propertyName[$i]) {
    $snakeCasedName .= '_'.$lcPropertyName;
} else {
    $snakeCasedName .= $lcPropertyName;
}
 ```